pkg/service/id.go

```diff
@@ -16,19 +16,52 @@ package service
 
 import (
 	"github.com/cilium/cilium/common/types"
+	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
 // AcquireID acquires a service ID
 func AcquireID(l3n4Addr types.L3n4Addr, baseID uint32) (*types.L3n4AddrID, error) {
-	return acquireGlobalID(l3n4Addr, baseID)
+	log.WithField(logfields.L3n4Addr, logfields.Repr(l3n4Addr)).Debug("Resolving service")
+
+	if enableGlobalServiceIDs {
+		return acquireGlobalID(l3n4Addr, baseID)
+	}
+
+	return acquireLocalID(l3n4Addr, baseID)
 }
 
 // GetID returns the L3n4AddrID that belongs to the given id.
 func GetID(id uint32) (*types.L3n4AddrID, error) {
-	return getGlobalID(id)
+	if enableGlobalServiceIDs {
```

Comment:
Instead of this model, why not use an interface  (e.g., ServiceIDAllocator), which implements some functions (getID, acquireID, etc.) and set the type of this interface based on the flag `enableGlobalServiceIDs` in some initialization function?

Reply:
I thought about it and just didn't see a clear immediate benefit besides use of interface for the purpose of using an interface. It's easy to change though so I'm happy to switch.

Reply:
No worries, it's not a blocking issue for me.